pkg/provider/vault/vault_test.go

```diff
@@ -250,6 +250,20 @@ MIIFkTCCA3mgAwIBAgIUBEUg3m/WqAsWHG4Q/II3IePFfuowDQYJKoZIhvcNAQELBQAwWDELMAkGA1UE
 				err: errors.New(errVaultStore),
 			},
 		},
+		"InvalidRetrySettings": {
```

Comment:
Would you add one more test case for valid retry settings?

Reply:
I need help for that, the code would be:
```go
		"ValidRetrySettings": {
			reason: "Should return a Vault provider with custom retry settings",
			args: args{
				store: makeSecretStore(func(s *esv1beta1.SecretStore) {
					s.Spec.RetrySettings = &esv1beta1.SecretStoreRetrySettings{
						MaxRetries:    pointer.To(int32(3)),
						RetryInterval: pointer.To("10m"),
					}
				}),
				ns:   "default",
				kube: clientfake.NewClientBuilder().Build(), // TODO here I need some magic instead of an empty object
			},
			want: want{
				err: nil,
			},
		},
```
but I need to know how to create a default but working kclient, or I get null pointers.
I don't want to copy it from another test or I'll introduce unrelated cross failures.

Reply:
Setting `corev1` and `newClientFunc` should do the work 🙂 

```
			args: args{
				store: makeSecretStore(func(s *esv1beta1.SecretStore) {
					s.Spec.RetrySettings = &esv1beta1.SecretStoreRetrySettings{
						MaxRetries:    pointer.To(int32(3)),
						RetryInterval: pointer.To("10m"),
					}
				}),
				ns:            "default",
				kube:          clientfake.NewClientBuilder().Build(),
				corev1:        utilfake.NewCreateTokenMock().WithToken("ok"),
				newClientFunc: fake.ClientWithLoginMock,
			},
```

Reply:
Here we go.